.github/workflows/build-and-test.yml

```diff
@@ -31,6 +32,11 @@ jobs:
       - name: create build environment
         run: cmake -E make_directory ${{ runner.workspace }}/_build
 
+      - name: setup lto
```

Comment:
This ensures the test cases work but is not required.

Reply:
it's good to leave it in then. i like that we're testing with it on :)

Reply:
LTO was not yet active. I fixed this now. However, static Ubuntu builds warn about ODR violations and indicate that `-fno-strict-aliasing` is necessary for `cxx03_test` to compile. Also `-Werror=lto-type-mismatch` is a problem. The compiler seems to be right here as `cxx03_test` should not be using any data structures that were compiled with C++11 ABI.

Reply:
Fixed this now as well. The warnings are suppressed.

Reply:
i'm not sure if warning suppression is appropriate. i think this is getting at what @LebedevRI has been suggesting: that the library, while it builds under c++11, should also build/run under c++03 for some platforms, and this ABI mismatch won't allow that (i think? or is it a matter of building cxx03_test differently to ensure no ABI mismatch?)

Reply:
ABI mismatch would cause linker errors. There's none though. Suppression is the same as `-Wno-odr` but for LTO which `cxx03_test` violates either way.

Reply:
@LebedevRI's point about `= delete` is not relevant here and probably does not make any sense anyway. I'm not a compiler engineer, but I don't see how a deleted symbol would have any implications on the ABI since the symbol cannot be used in the first place.